app/models/user.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+class User
+  attr_accessor :id
+  attr_writer :identity
+
+  def self.find_by_username(username)
+    @valid_users ||= Ettin.for(Ettin.settings_files('config', Rails.env))[:users]
```

Comment:
I also think `find_by_username` probably isn't the right place to do the authorization; for one thing, right now it's only called by `SessionController#create`, which won't be used in production.

Maybe just have `SessionController#create` call `User.new`, since we don't actually have persisted users?

Then, we could do the authorization (for now anyway) in another `before_action` in `ApplicationController` - I think the shape should look something like:

```ruby
before_action :authorize!

def authorize!
  raise NotAuthorizedError unless Otis.config.valid_users.include? current_user.username
end
```

That's assuming we move the Ettin load to `config/application.rb`, where it will look like:

```ruby
module Otis
  class << self
    def config
      @config ||= Ettin.for(Ettin.settings_files("config", Rails.env))
    end
  end

  # eager load
  config

  class Application < Rails::Application
    # usual rails config goes here
    ...
  end
end
```